enhancements/insights/pulling-sca-certs-from-ocm.md

```diff
@@ -0,0 +1,156 @@
+---
+title: pulling-and-exposing-sca-certs-from-ocm
+authors:
+  - "@tremes"
+reviewers:
+  - "@sbose78"
+  - "@inecas"
+  - "@petli-openshift"
+  - "@bparees"
+  - "@dhellman"
+  - "@mfojtik"
+  - "@adambkaplan"
+approvers:
+  - "@sbose78"
+  - "@bparees"
+  - "@dhellman"
+creation-date: 2021-03-04
+last-updated: 2021-08-10
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Insights Operator pulling and exposing SCA certs from the OCM API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement will enable the Insights Operator to pull the data (SCA certs)
+from the OCM (OpenShift Cluster Manager) API. The data will be exposed by the Insights Operator
+in the OpenShift API to allow users to use them when consuming and building container images
+on the platform.
+
+## Motivation
+
+Users could consume RHEL content and container images using the RHEL subscription in the OpenShift 3.x.
+In the OpenShift 4, this is no longer possible because the Red Hat Enterprise Linux Core OS (RHCOS) does not
+provide any attached subscription. This enhancement is to provide users the Simple Content Access (SCA) certs
+from Red Hat Subscription Manager (RHSM).
+
+### Goals
+
+- Extend the Insights Operator config with an OCM API URL to be able to query the data
+- Periodically pull the data from the OCM API and expose it in the OpenShift API
+- This is an opt-in feature by a cluster user and might be moved to a different OCP component in the future
+
+### Non-Goals
+
+- Insights Operator providing any transformation or post-processing of the SCA certs pulled
+  from the OCM API
+
+## Proposal
+
+### Why is it in the Insights Operator?
+
+The Insights Operator is now the only OCP component that connects an OpenShift cluster to a Red Hat subscription experience (console.redhat.com APIs). The consumers of the SCA certs are not only builds, but also shared resources, such as the CSI driver.
+
+### User Stories
+
+#### Consume SCA certs exposed in the API
+
+As an OpenShift user
+I want to consume SCA certs to be able to consume RHEL content and to build
+corresponding container images.
+
+### Risks and Mitigations
+
+#### OCM API is down
+
+Risk: OCM API is down or doesn't provide up to date data.
+
+Risk: Insights Operator is unable to expose/update the data in the OpenShift API
+
+Mitigation: Introduce a new state in the Insights Operator (e.g "SCADataDegraded") and
+create a new alert based on this new state.
```

Comment:
this is outdated, right?  i think there was agreement to just use the degraded state.

since it's a techpreview feature, people can't upgrade anyway, but we can use this to gather data about whether the operator is going degraded in customer clusters frequently, and what the implications of that are, before we GA the feature.

We should definitely ensure we've gathered some data on how often this operator is going degraded due to this feature before we GA the feature.

Reply:
OK makes sense to me. Perhaps we should explicitly define the HTTP error states when the IO goes degraded (after some threshold/retries), because for example 404 means that the org didn't allow this feature (in OCM or RHSM I guess) so there's no reason to degrade it.

Reply:
yes that would be good.

Reply:
I added it to the mitigation description.